xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/html/HTMLUtils.java

```diff
@@ -113,7 +113,7 @@ public String escapeElementEntities(String text)
             String result = super.escapeElementEntities(text);
 
             // "\r" characters are automatically transformed in &#xD; but we want to keep the original \r there.
-            return cleanAmpersandEscape(result).replaceAll("&#xD;", "\r");
+            return result.replaceAll("&#xD;", "\r");
```

Comment:
This has to be double checked, see discussion in https://github.com/xwiki/xwiki-commons/pull/149/files#r778652172

Reply:
I believe it's ok as it's related to the bad escape that was fixed in: https://sourceforge.net/p/htmlcleaner/bugs/221/